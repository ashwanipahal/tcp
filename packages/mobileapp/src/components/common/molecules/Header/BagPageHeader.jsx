import React from 'react';
import { connect } from 'react-redux';
import { getLocator, isGymboree } from '@tcp/core/src/utils';
import ToastContainer from '@tcp/core/src/components/common/atoms/Toast/container/Toast.container.native';
import {
  SafeAreaViewStyle,
  BrandIcon,
  CloseIcon,
  BrandIconSection,
  CloseIconTouchable,
  CloseContainer,
  BagPageContainer,
} from './Header.style';

// @flow
type Props = {
  navigation: object,
};

/**
 * This component creates Mobile Header
 * @param {*} props Props passed from Header screen
 */

const closeIcon = require('@tcp/core/src/assets/close.png');
const tcpIcon = require('../../../../assets/images/tcp/tcpLaunchImage.png');
const gymIcon = require('../../../../assets/images/gymboree/gymboreeLaunchImage.png');

/**
 * This component creates Mobile Header.
 * 1. To Manage the store locator
 * 2. To Navigate the cart page & show cart quantity
 * 3. To show the welcome text for guest user
 *     and shoe the name fro register user
 */
class BagPageHeader extends React.PureComponent<Props> {
  closeIconAction = () => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  render() {
    const { isPayPalWebViewEnable } = this.props;
    return (
      <SafeAreaViewStyle>
        <ToastContainer />
        {!isPayPalWebViewEnable && (
          <BagPageContainer data-locator={getLocator('global_bagpageheaderpanel')}>
            <BrandIconSection>
              <BrandIcon
                source={isGymboree() ? gymIcon : tcpIcon}
                data-locator={getLocator('global_bagpageheaderpanelbrandicon')}
                accessibilityRole="image"
              />
            </BrandIconSection>
            <CloseContainer>
              <CloseIconTouchable onPress={this.closeIconAction}>
                <CloseIcon
                  source={closeIcon}
                  data-locator={getLocator('global_bagpageheaderpanelcloseicon')}
                  accessibilityRole="button"
                />
              </CloseIconTouchable>
            </CloseContainer>
          </BagPageContainer>
        )}
      </SafeAreaViewStyle>
    );
  }
}

const mapStateToProps = state => {
  return {
    labels: state.Labels.global && state.Labels.global.header,
    isPayPalWebViewEnable:
      state.CartPageReducer.getIn(['uiFlags', 'isPayPalWebViewEnable']) || false,
  };
};

export default connect(mapStateToProps)(BagPageHeader);
export { BagPageHeader as BagPageHeaderVanilla };
