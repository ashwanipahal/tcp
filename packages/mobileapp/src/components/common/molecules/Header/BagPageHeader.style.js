import styled from 'styled-components/native';

export const BagPageBackContainer = styled.TouchableOpacity`
  margin-top: ${props => props.theme.spacing.ELEM_SPACING.MED};
  margin-left: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  width: 100%;
  border-bottom-color: ${props => props.theme.colorPalette.gray[500]};
  border-bottom-width: 1;
`;

export default {
  BagPageBackContainer,
};
