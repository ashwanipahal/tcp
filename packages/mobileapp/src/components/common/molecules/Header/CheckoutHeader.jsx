import React from 'react';
import { Text } from 'react-native';
import { connect } from 'react-redux';
import { getLocator } from '@tcp/core/src/utils';
import {
  MessageContainer,
  StoreContainer,
  SafeAreaViewStyle,
  ImageColor,
  CheckoutHeaderContainer,
  CheckoutHeaderTextSection,
  BackIcon,
  BackIconTouchable,
} from './Header.style';

// @flow
type Props = {
  navigation: object,
};

/**
 * This component creates Mobile Header
 * @param {*} props Props passed from Header screen
 */

const leftIcon = require('../../../../assets/images/carrot-large-left.png');

/**
 * This component creates Mobile Header.
 * 1. To Manage the store locator
 * 2. To Navigate the cart page & show cart quantity
 * 3. To show the welcome text for guest user
 *     and shoe the name fro register user
 */
class CheckoutHeader extends React.PureComponent<Props> {
  render() {
    const { navigation } = this.props;

    return (
      <SafeAreaViewStyle>
        <CheckoutHeaderContainer data-locator={getLocator('global_headerpanel')}>
          <MessageContainer>
            <StoreContainer>
              <BackIconTouchable
                onPress={() => {
                  navigation.pop();
                }}
              >
                <BackIcon
                  source={leftIcon}
                  style={ImageColor}
                  data-locator={getLocator('global_headerpanelcollapsedicon')}
                  accessibilityRole="button"
                />
              </BackIconTouchable>
            </StoreContainer>
          </MessageContainer>
          <CheckoutHeaderTextSection>
            <Text className="product-name">CHECKOUT</Text>
          </CheckoutHeaderTextSection>
        </CheckoutHeaderContainer>
      </SafeAreaViewStyle>
    );
  }
}

const mapStateToProps = state => {
  return {
    labels: state.Labels.global && state.Labels.global.header,
  };
};

export default connect(mapStateToProps)(CheckoutHeader);
export { CheckoutHeader as CheckoutHeaderVanilla };
