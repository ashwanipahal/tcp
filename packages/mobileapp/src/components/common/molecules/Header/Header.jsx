import React from 'react';
import { connect } from 'react-redux';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import { getLocator } from '@tcp/core/src/utils';
import HeaderPromo from '../HeaderPromo/HeaderPromo';
import { readCookieMobileApp } from '../../../../utils/utils';

import {
  Container,
  MessageContainer,
  StoreContainer,
  CartContainer,
  Icon,
  RoundView,
  SafeAreaViewStyle,
  TextStyle,
  BackgroundView,
  CartIconView,
  ImageColor,
  HeaderPromoContainer,
  Touchable,
} from './Header.style';

// @flow
type Props = {
  labels: object,
  headerPromo: Array,
  navigation: object,
};

/**
 * This component creates Mobile Header
 * @param {*} props Props passed from Header screen
 */

const downIcon = require('../../../../assets/images/carrot-small-down.png');
const upIcon = require('../../../../assets/images/carrot-small-up.png');
const cartIcon = require('../../../../assets/images/empty-bag.png');

/**
 * This component creates Mobile Header.
 * 1. To Manage the store locator
 * 2. To Navigate the cart page & show cart quantity
 * 3. To show the welcome text for guest user
 *     and shoe the name fro register user
 */
class Header extends React.PureComponent<Props> {
  /**
   * To manage the state of icons on the
   * basis of expand & collaps .
   */
  constructor(props) {
    super(props);
    const CART_ITEM_COUNTER = 'cartItemsCount';
    this.state = {
      isDownIcon: false,
      cartVal: parseInt(readCookieMobileApp(CART_ITEM_COUNTER) || 0, 10),
    };
  }

  /**
   * This function validate the iconView.
   */
  validateIcon = () => {
    const { isDownIcon } = this.state;
    this.setState({
      isDownIcon: !isDownIcon,
    });
  };

  renderPromo = () => {
    const { headerPromo } = this.props;
    if (headerPromo) {
      return (
        <HeaderPromoContainer>
          <HeaderPromo headerPromo={headerPromo} />
        </HeaderPromoContainer>
      );
    }
    return null;
  };

  render() {
    const { isDownIcon, cartVal } = this.state;
    let headerLabels = {
      lbl_header_storeDefaultTitle: '',
      lbl_header_welcomeMessage: '',
    };

    const { labels } = this.props;
    if (labels) {
      headerLabels = labels;
    }

    return (
      <SafeAreaViewStyle>
        <Container data-locator={getLocator('global_headerpanel')}>
          <MessageContainer>
            <BodyCopy
              fontFamily="secondary"
              fontSize="fs14"
              textAlign="center"
              color="black"
              fontWeight="semibold"
              text={headerLabels.lbl_header_welcomeMessage}
              data-locator={getLocator('global_headerpanelwelcometext')}
            />
            <StoreContainer onPress={this.validateIcon}>
              <BodyCopy
                fontFamily="secondary"
                fontSize="fs12"
                textAlign="center"
                color="text.primary"
                fontWeight="regular"
                text={headerLabels.lbl_header_storeDefaultTitle}
                data-locator={getLocator('global_findastoretext')}
                accessibilityText="Drop Down"
              />
              {isDownIcon ? (
                <Icon
                  source={upIcon}
                  style={ImageColor}
                  data-locator={getLocator('global_headerpanelexpandedicon')}
                />
              ) : (
                <Icon
                  source={downIcon}
                  style={ImageColor}
                  data-locator={getLocator('global_headerpanelcollapsedicon')}
                />
              )}
            </StoreContainer>
          </MessageContainer>
          <CartContainer>
            <Touchable
              accessibilityRole="button"
              onPress={() => {
                // eslint-disable-next-line react/destructuring-assignment
                this.props.navigation.navigate('BagPage');
              }}
            >
              <CartIconView
                source={cartIcon}
                data-locator={getLocator('global_headerpanelbagicon')}
              />
              <BackgroundView />
              <RoundView />
              <BodyCopy
                text={cartVal}
                color="white"
                style={TextStyle}
                fontSize="fs10"
                data-locator={getLocator('global_headerpanelbagitemtext')}
                accessibilityText="Mini bag with count"
              />
            </Touchable>
          </CartContainer>
        </Container>
        {this.renderPromo()}
      </SafeAreaViewStyle>
    );
  }
}

const mapStateToProps = state => {
  return {
    labels: state.Labels.global && state.Labels.global.header,
    headerPromo: state.Header && state.Header.promoTextBannerCarousel,
  };
};

export default connect(mapStateToProps)(Header);
export { Header as HeaderVanilla };
