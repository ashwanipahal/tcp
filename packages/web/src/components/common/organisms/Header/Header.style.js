import { css } from 'styled-components';

export default css`
  .header-topnav {
    background-color: ${props => props.theme.colors.PRIMARY.PALEGRAY};

    & > div:first-child {
      margin-left: 6px;
    }

    .header-topnav__brand-tabs {
      padding-right: 0;
    }

    .header-topnav__promo-area {
      text-align: center;

      @media ${props => props.theme.mediaQuery.smallMax} {
        display: none;
      }
    }
    .header-topnav__track-order {
      text-align: right;

      @media ${props => props.theme.mediaQuery.mediumMax} {
        padding-right: 0;
      }
    }
  }
  .header-brand {
    box-sizing: border-box;
    height: 129px;
    padding-top: 31px;
    text-align: center;

    @media ${props => props.theme.mediaQuery.mediumMax} {
      .tcp-logo {
        display: none;
      }
    }
  }
  .dummy-nav {
    color: ${props => props.theme.colors.PRIMARY.DARK};
    cursor: pointer;
    display: flex;
    font-family: ${props => props.theme.fonts.secondaryFontFamily};
    font-size: ${props => props.theme.fonts.fontSize.nav}px;
    font-weight: ${props => props.theme.fonts.fontWeight.semiBold};
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    text-transform: uppercase;
    padding: 10px 0;

    @media ${props => props.theme.mediaQuery.mediumMax} {
      display: none;
    }
  }
  .header-promo {
    background-color: ${props => props.theme.colors.PRIMARY.PALEGRAY};
    border-top: 1px solid ${props => props.theme.colors.PRIMARY.LIGHTGRAY};
    box-sizing: border-box;
    height: 60px;
    padding: 20px 0;
    text-align: center;
    text-transform: uppercase;
  }
  .header-loyalty {
    background-color: ${props => props.theme.colors.PRIMARY.LIGHTGRAY};
    padding: 36px 0;
    text-align: center;
    text-transform: uppercase;
  }
`;
