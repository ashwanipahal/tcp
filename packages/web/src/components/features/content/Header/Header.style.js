import { css } from 'styled-components';

export default css`
  margin-bottom: 8px;

  @media ${props => props.theme.mediaQuery.large} {
    margin-bottom: 24px;
  }

  .header-middle-nav-bar {
    border-bottom: 1px solid ${props => props.theme.colorPalette.text.disabled};
  }

  .checkout-pages & {
    display: none;
  }

  .skip-main {
    padding: 6px;
    position: absolute;
    top: -40px;
    left: 0px;
    color: ${props => props.theme.colorPalette.white};
    border-right: 1px solid ${props => props.theme.colorPalette.white};
    border-bottom: 1px solid ${props => props.theme.colorPalette.white};
    border-bottom-right-radius: 8px;
    background: ${props => props.theme.colors.BRAND.PRIMARY};
    -webkit-transition: top 1s ease-out;
    transition: top 1s ease-out;
    z-index: ${props => props.theme.zindex.zEnlargedImage};
  }

  .skip-main:focus,
  .skip-main:active {
    position: absolute;
    left: 0px;
    top: 0px;
    outline-color: transparent;
    -webkit-transition: top 0.1s ease-in;
    transition: top 0.1s ease-in;
  }
  .top-promo-bannner {
    display: none;
    @media ${props => props.theme.mediaQuery.large} {
      display: block;
    }
  }
`;
