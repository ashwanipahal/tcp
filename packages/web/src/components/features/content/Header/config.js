export default {
  tcp: {
    alt: "The Children's Place",
    dataLocator: 'global_TCPlink',
    imgSrc: '/static/images/tcp-logo.svg',
    title: "The Children's Place",
  },
  gym: {
    alt: 'Gymboree',
    dataLocator: 'global_Gymboreelink',
    imgSrc: '/static/images/gymboree-logo.svg',
    title: 'Gymboree',
  },
  l1HoverDelay: 150,
};
