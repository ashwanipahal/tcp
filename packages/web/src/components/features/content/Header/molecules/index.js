import HeaderPromo from './HeaderPromo';
import HeaderTopNav from './HeaderTopNav';
import BrandTabs from './BrandTabs';
import CountrySelector from './CountrySelector';
import PromotionalArea from './PromotionalArea';
import HeaderMiddleNav from './HeaderMiddleNav';

export { HeaderPromo, HeaderTopNav, BrandTabs, CountrySelector, PromotionalArea, HeaderMiddleNav };
