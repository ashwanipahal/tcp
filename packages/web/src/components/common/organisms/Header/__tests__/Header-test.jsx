import React from 'react';
import { shallow } from 'enzyme';
import Header from '../views';

describe('Header component', () => {
  it('renders correctly', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
