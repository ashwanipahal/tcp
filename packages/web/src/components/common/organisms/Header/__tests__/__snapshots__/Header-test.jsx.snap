// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header>
  <Styled(HeaderTopNav)
    className="header-topnav"
    dataTopNav={
      Object {
        "composites": Object {},
      }
    }
  />
  <styled.div
    className="header-brand"
  >
    <Styled(Row)>
      <Styled(Col)
        className="header-brand__home-logo"
        colSize={
          Object {
            "large": 12,
            "medium": 8,
            "small": 6,
          }
        }
      >
        <Styled(HomeLogo)
          alt="The Children's Place"
          className="header-brand__home-logo--brand"
          dataLocator="global_TCPlink"
          imgSrc="/static/images/tcp-logo.svg"
        />
      </Styled(Col)>
    </Styled(Row)>
  </styled.div>
  <styled.div
    className="header-nav"
  >
    <Styled(Row)>
      <Styled(Col)
        className="header-nav__nav-row"
        colSize={
          Object {
            "large": 12,
            "medium": 0,
            "small": 0,
          }
        }
      >
        <styled.div
          className="dummy-nav"
        >
          <div>
            Girl
          </div>
          <div>
            Toddler Girl
          </div>
          <div>
            Boy
          </div>
          <div>
            Toddler Boy
          </div>
          <div>
            Baby
          </div>
          <div>
            Shoes
          </div>
          <div>
            Accessories
          </div>
          <div>
            Trending
          </div>
          <div>
            Gift Cards
          </div>
          <div>
            Clearance
          </div>
        </styled.div>
      </Styled(Col)>
    </Styled(Row)>
  </styled.div>
  <Styled(HeaderPromo)
    className="header__promo-area--mobile"
    dataPromo={Object {}}
    mobileMarkup={true}
  />
  <Styled(HeaderPromo)
    className="header__promo-area--desktop"
    dataPromo={Object {}}
  />
  <styled.div
    className="header-loyalty"
  >
    <Styled(Row)>
      <Styled(Col)
        className="header-loyalty__promo-loyalty"
        colSize={
          Object {
            "large": 12,
            "medium": 8,
            "small": 6,
          }
        }
      >
        Loyalty Promo banners
      </Styled(Col)>
    </Styled(Row)>
  </styled.div>
</header>
`;
